Android/quickstart-kotlin/app/src/main/java/com/example/immersivereadersdk/MainActivity.kt

```diff
@@ -0,0 +1,251 @@
+
+        val data = Content(irTitle.text.toString(), chunks)
+
+        // Options passed to the Immersive Reader
+        val allowFullscreen = true
+        val customDomain = ""
+        val hideExitButton = false
+        val onExit = appUtilities::exitCallback
+        val uiLang = "en"
+        val uiZIndex = 0
+        val timeoutInMilliseconds = 15000
+        val options = Options(uiLang, timeoutInMilliseconds, uiZIndex, onExit, customDomain, allowFullscreen, hideExitButton)
+
+        var token: String
+
+        runBlocking{
+            val resp = async { getImmersiveReaderTokenAsync() }
+            token = resp.await()
+            val jsonResp = JSONObject(token)
+            loadImmersiveReaderWebView(exampleActivity, jsonResp.getString("access_token"), subdomain, data, options)
+        }
+    }
+
+    private suspend fun getImmersiveReaderTokenAsync(): String {
+        return getToken()
+    }
+
+    @Throws(IOException::class)
+    fun getToken(): String {
+        val clientId = dotEnv["CLIENT_ID"]
+        val clientSecret = dotEnv["CLIENT_SECRET"]
+        val tenantId = dotEnv["TENANT_ID"]
+        val tokenUrl = URL("https://login.windows.net/$tenantId/oauth2/token")
+        val form = "grant_type=client_credentials&resource=https://cognitiveservices.azure.com/&client_id=$clientId&client_secret=$clientSecret"
+
+        val connection = tokenUrl.openConnection() as HttpURLConnection
+        connection.requestMethod = "POST"
+        connection.setRequestProperty("content-type", "application/x-www-form-urlencoded")
+        connection.doOutput = true
+
+        val writer = DataOutputStream(connection.outputStream)
+        writer.writeBytes(form)
+        writer.flush()
+        writer.close()
+
+        val responseCode = connection.responseCode
+
+        if (responseCode == HTTP_OK) {
+            val readerIn = BufferedReader(InputStreamReader(connection.inputStream))
+            var inputLine = readerIn.readLine()
+            val response = StringBuffer()
+
+            do {
+                response.append(inputLine)
+            } while (inputLine.length < 0)
+            readerIn.close()
+
+            // Return token
+            return response.toString()
+        } else {
+            val responseError = Error(code = "BadRequest", message = "There was an error getting the token.")
+            throw IOException(responseError.toString())
+        }
+    }
+    
+    data class Content(var title: String,
+                       var chunks: List<Chunk>)
+
+    data class Chunk(var content: String,
+                     var lang: String,
+                     var mimeType: String)
+
+    // Only includes Immersive Reader options relevant to Android apps.
+    // For a complete list see https://github.com/microsoft/immersive-reader-sdk/blob/master/js/src/options.ts
+    data class Options(var uiLang: String, // Language of the UI, e.g. en, es-ES (optional). Defaults to browser language if not specified.

```

Comment:
did you decide that we can't treat options as a simple dictionary that we can serialize into json when we sent it through the message, and deserialize into a dictionary in the html page? otherwise we would have to update this class every time we add an option

Reply:
What's the reason we would have to update the class? The reason why I ask is because I think having well defined classes to map json to is the correct way to go.

Reply:
Not sure what you mean here. At the moment the owness is on the user to maintain these classes because they're private and cannot be read from the SDK.